### ISSUE TITLE
Scan for PHP parse errors in all supported PHP versions

### ISSUE THREAD

**Issue by jrfnl** [wptide/wptide#138]
Extracted from https://github.com/wptide/wptide/issues/137#issuecomment-399782731

> It would be good to scan for parse errors in all PHP versions supported by the plugin/theme.
>
> Just running `php -l` over all files in the plugin/theme (possibly excluding a vendor directory) on all supported PHP versions should be able to take care of that.

Ref: http://php.net/manual/en/features.commandline.options.php

Potentially running `php -f` (parse and execute) could also yield interesting results, though may throw too many false positives.

**Comment by hellofromtonya** [comment]
Note: `php -l` is included PHPCS' [`Generic.PHP.Syntax` ruleset](https://github.com/squizlabs/PHP_CodeSniffer/blob/87ba8779e7335a491f14917d1604e81dfe3a2890/src/Standards/Generic/Sniffs/PHP/SyntaxSniff.php#L64): 

```
$cmd      = escapeshellcmd($this->phpPath)." -l -d display_errors=1 -d error_prepend_string='' $fileName 2>&1";
```

which is included in [WPCS WordPress Extra ruleset](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/develop/WordPress-Extra/ruleset.xml#L149):

```
	<!-- Check for PHP Parse errors.
		 https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/522 -->
	<rule ref="Generic.PHP.Syntax"/>
```

**Comment by jrfnl** [comment]
@hellofromtonya Correct, but it was only included as too many people don't know how to run `php -l` :cry: 

Also: "all" PHPCS/WPCS/PHPCompatibility sniffs should give the same results independently of the PHP version they are run on (though higher is better), so only one run of PHPCS on a high PHP version is warranted to get the sniff results.

The results from the `Generic.PHP.Syntax` sniff, however, *will* differ between PHP versions.

So, as "we" actually do know what we are doing, if the syntax sniff is included in an external ruleset used for Tide, it should be `exclude`d, the sniffs should only be run once on a high PHP version and linting should be done separately against all PHP versions supported by a plugin/theme.

**Comment by hellofromtonya** [comment]
>Correct, but it was only included as too many people don't know how to run `php -l`.

True. I know a lot of devs don't know it exists.

>The results from the Generic.PHP.Syntax sniff, however, will differ between PHP versions.

100% agree.

>it should be excluded, the sniffs should only be run once on a high PHP version and linting should be done separately against all PHP versions supported by a plugin/theme.

Agreed.

**Comment by jeffpaul** [comment]
Per [bugscrub](https://wordpress.slack.com/archives/C7TK8FBUJ/p1535488079000100) today, we're moving this to `Future Release`.

**Comment by jrfnl** [comment]
I can highly recommend using the [`Parallel-Lint`](https://github.com/JakubOnderka/PHP-Parallel-Lint) package for speedy linting, works on PHP 5.4 and higher, so this might be a good option for when this issue gets addressed.
